lib/ansible/executor/task_queue_manager.py

```diff
@@ -81,6 +82,30 @@ def __init__(self, *args, **kwargs):
         kwargs['ctx'] = multiprocessing_context
         super(FinalQueue, self).__init__(*args, **kwargs)
 
+    def _on_queue_feeder_error(self, e, obj):
+        """This isn't really a public method, but it is overridden by concurrent.futures
+
+        So we'll pretend it's safe enough to override here too, in lieu of an actual public
+        API to achieve this goal.
+        """
+        if isinstance(obj, TaskResult):
+            display.debug(f'failed to send task result ({e}), sending surrogate result')
+            self.send_task_result(
+                obj._host,
+                obj._task,
```

Comment:
These are already not objects, it's misleading, but they are `Host.name` and `Task._uuid` here, and we reconcile this on the other side of the queue.  There is only 1 case where this is not true, and it's explicitly documented in code (implicit async_status for `poll > 0`)

Reply:
Yeah, coulda sworn we were already doing that conversion- it was because the first breakpoint I hit in there was in `meta`, which is in the controller process and still passing the full objects around. Verified that the worker-side values are `str`.

Reply:
So yeah, since these are all supposedly composed of primitive builtin types, this should be fine...